packages/cli/src/lib/bundler/server.ts

```diff
@@ -139,58 +161,99 @@ export async function serveBundle(options: ServeOptions) {
     additionalEntryPoints: detectedModulesEntryPoint,
   });
 
-  const compiler = webpack(config);
-
-  server = new WebpackDevServer(
-    {
-      hot: !process.env.CI,
-      devMiddleware: {
-        publicPath: config.output?.publicPath as string,
-        stats: 'errors-warnings',
+  if (process.env.EXPERIMENTAL_VITE) {
+    server = await vite.createServer({
+      define: {
+        global: 'globalThis',
+        'process.argv': JSON.stringify(process.argv),
+        'process.env.APP_CONFIG': JSON.stringify(cliConfig.frontendAppConfigs),
+      },
+      resolve: {
+        alias: {
+          'node-fetch': 'cross-fetch',
+        },
       },
-      static: paths.targetPublic
-        ? {
-            publicPath: config.output?.publicPath as string,
-            directory: paths.targetPublic,
-          }
-        : undefined,
-      historyApiFallback: {
-        // Paths with dots should still use the history fallback.
-        // See https://github.com/facebookincubator/create-react-app/issues/387.
-        disableDotRule: true,
-
-        // The index needs to be rewritten relative to the new public path, including subroutes.
-        index: `${config.output?.publicPath}index.html`,
+      plugins: [
+        react(),
+        vitePluginSvgr(),
```

Comment:
I think we could skip this for now if it causes trouble tbh, it's just for the `.icon.svg` support right? Or is it to be able to import `.svg` as an external resource ref too?

Reply:
Ah, outdated one from earlier, I assume this is what you did? :p

Reply:
ye